waggle-dance-core/src/test/java/com/hotels/bdp/waggledance/client/compatibility/HiveCompatibleThriftHiveMetastoreIfaceFactoryTest.java

```diff
@@ -162,8 +183,154 @@ public void get_foreign_keys() throws Exception {
     ForeignKeysRequest foreignKeysRequest = new ForeignKeysRequest(null, null, DB_NAME, TABLE_NAME);
     when(delegate.get_foreign_keys(foreignKeysRequest)).thenThrow(new TApplicationException("Error"));
     ForeignKeysResponse foreignKeysResponse = thriftHiveMetastoreIface.get_foreign_keys(foreignKeysRequest);
-    assertThat(foreignKeysResponse, is(new ForeignKeysResponse(Collections.emptyList())));
+    assertThat(foreignKeysResponse, is(new ForeignKeysResponse(emptyList())));
     verify(delegate).get_table(DB_NAME, TABLE_NAME);
   }
 
+  @Test
+  public void noSuchMethodInCompatibilityLayerHandling() throws Exception {
+    CloseableThriftHiveMetastoreIface thriftHiveMetastoreIface = factory.newInstance(delegate);
+    when(delegate.get_database(DB_NAME)).thenThrow(new TApplicationException("Error"));
```

Comment:
I'd be curious, you can make the mock fail on anything, but why are you making it fail on `get_database`?

Reply:
Thanks.